subprojects/packagefiles/mpdecimal/tests/gettests.py

```diff
@@ -0,0 +1,23 @@
+#!python3
+
+import hashlib
+import io
+import os
+import shutil
+import sys
+import urllib.request
+import zipfile
+
+SHA256 = 'b70a224cd52e82b7a8150aedac5efa2d0cb3941696fd829bdbe674f9f65c3926'
+
+src, dst, *download = sys.argv[1:]
+
+if not os.path.isfile(os.path.join(dst, 'baseconv.decTest')):
+    shutil.copytree(src, dst, copy_function=shutil.copyfile, dirs_exist_ok=True)
+
+if download and not os.path.isfile(os.path.join(dst, 'add.decTest')):
+    with urllib.request.urlopen('http://speleotrove.com/decimal/dectest.zip') as f:
+        data = io.BytesIO(f.read())
+        assert hashlib.sha256(data.getbuffer()).hexdigest() == SHA256
```

Comment:
Python assert is not a functionality for checking whether or not to do something. It's a functionality for running debug code that double-checks whether the program is performing as expected. It can be optimized out to nothing.

Just use if/raise. :)

Reply:
It would be technically more correct as a `raise ValueError` than as a `raise AssertionError` although, indeed, either one promotes this from "logically impossible" to "error checking".

So that's acceptable.